spirv_msl.cpp

```diff
@@ -8193,6 +8824,9 @@ string CompilerMSL::entry_point_args_argument_buffer(bool append_comma)
 				buffer_binding = i;
 		}
 
+		// Allow the caller to specify an offset for argument buffer binding slots
+		buffer_binding += msl_options.argument_buffer_offset;
```

Comment:
This is actually kinda redundant. You can use add_msl_resource_binding to pick where IABs themselves are placed.